### PR TITLE
Notify owners about organization deletion

### PR DESCRIPTION
**Pull request by mattrobenolt**
And prompt them to cancel and restore the organization.

@getsentry/infrastructure @getsentry/ui 

Help on messaging here would be nice since what I put is likely garbage.

Web UI:
image

Email:
image

TODO:
- [x] Tests
- [x] Better copy

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/4116)

### CODE REVIEW

File: src/sentry/models/organization.py [hunk 1]
```diff
@@ -150,23 +158,27 @@ def get_audit_log_data(self):
             'default_role': self.default_role,
         }
 
+    def get_owners(self):
+        from sentry.models import User
+        return User.objects.filter(
+            sentry_orgmember_set__role=roles.get_top_dog().id,
```
Comment (JTCunning) [inline src/sentry/models/organization.py]: Favorite.

File: src/sentry/web/frontend/debug/mail.py [hunk 2]
```diff
@@ -632,3 +632,29 @@ def recover_account(request):
...
         },
     ).render(request)
+
+
+@login_required
+def org_delete_confirm(request):
+    from sentry.models import AuditLogEntry
+
+    org = Organization.get_default()
+    entry = AuditLogEntry(
```
Comment (JTCunning) [inline src/sentry/web/frontend/debug/mail.py]: += `request=request,`

**Comment by ckj** [comment]
My stab at copy with one clear action that will work in both the app and email:

## Organization queued for deletion

The **Sentry** organization has been scheduled for deletion by:

```
User: Matt Robenolt
IP:lolcalhost
Date: Sept. 9, 2016, 11:46 p.m.
```

This irreversible deletion will take place at **Sept. 10, 2016, 11:46 p.m.** and will permanently remove all associated data including events, projects, and team members.

#### If this was unintentional and you would like to cancel the deletion:

[Restore Organization]

File: src/sentry/web/frontend/debug/mail.py (hunk 2 above)
Comment (mattrobenolt) [inline src/sentry/web/frontend/debug/mail.py]: Not needed here.

File: src/sentry/web/frontend/restore_organization.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        ).update(status=OrganizationStatus.VISIBLE)
+        if updated:
+            self.create_audit_entry(
+                request=request,
+                organization=organization,
+                target_object=organization.id,
+                event=AuditLogEntryEvent.ORG_RESTORE,
+                data=organization.get_audit_log_data(),
+            )
+            delete_logger.info('object.delete.canceled', extra={
```
Comment (JTCunning) [inline src/sentry/web/frontend/restore_organization.py]: TIL it's normal to have both `canceled` and `cancelled` in English.

**Comment by JTCunning** [comment]
Reviewed 8 of 12 files at r1.
Review status: 8 of 12 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/4116)*

File: src/sentry/web/frontend/restore_organization.py (hunk 3 above)
Comment (mattrobenolt) [inline src/sentry/web/frontend/restore_organization.py]: YUP

**Comment by mattrobenolt** [comment]
Alright, tests have been written, and removed "WIP" tag. :)

**Comment by dcramer** [comment]
fix yo linters yo

**Comment by mattrobenolt** [comment]
@dcramer fixed

**Review by dcramer — APPROVED** [review]
🚀

**Pull request merged**